rfc/replication.md

```diff
@@ -0,0 +1,125 @@
+  "aud": "did:web:upload.service.example.com",
+  "att": [
+    {
+      "with": "did:key:zAliceSpace",
+      "can": "space/blob/replicate",
+      "nb": {
+        /** The blob that MUST be replicated. */
+        "blob": {
+          "digest": { "/": { "bytes": "..." } },
+          "size": 1234
+        },
+        /** Number of replicas to ensure. */
+        "replicas": 2,
+        /** A location commitment indicating where the blob MUST be fetched from. */
+        "location": { "/": "bafy..locationCommitment" }
+      }
+    }
+  ],
+  "prf": [],
+  "sig": "..."
+}
+```
+
+It is RECOMMENDED that the location commitment is included in the invocation.
+
+The receipt for `space/blob/replicate` includes effects (async tasks) for `blob/replica/transfer`. Successful completion of the `blob/replica/transfer` task indicates the replication target has transferred and stored the blob. The number of `blob/replica/transfer` tasks corresponds directly to number of replicas requested.
+
+Each replication task MUST target a _different_ storage node and they MUST NOT target the original upload target.
+
+The upload service MUST select storage node(s) and allocate replication space when the `space/blob/replicate` invocation is received. The upload service allocates replication space on storage nodes by issuing a `blob/replica/allocate` invocation:
+
+```json5
+{
+  "iss": "did:web:upload.service.example.com",
+  "aud": "did:key:zStorageNode",
+  "att": [
+    {
+      "with": "did:key:zStorageNode",
+      "can": "blob/replica/allocate",
+      "nb": {
+        /** The blob that was must be replicated. */
+        "blob": {
+          "digest": { "/": { "bytes": "..." } },
+          "size": 1234
+        },
+        /** DID of the space the blob has been allocated to. */
+        "space": { "/": { "bytes": "..." } },
+        /** A location commitment indicating where the blob MUST be fetched from. */
+        "location": { "/": "bafy..locationCommitment" },
+        /** The `space/blob/replicate` invocation that caused this allocation. */
+        "cause": { "/": "bafy..replicate" }
+      }
+    }
+  ],
+  "prf": [],
+  "sig": "..."
+}
+```
+
+The `blob/replica/allocate` task receipt includes an async task that will be performed by the storage node - `blob/replica/transfer`. The `blob/replica/transfer` task is completed when the storage node has transferred the blob from its location to the storage node.
+
+A `blob/replica/transfer` task takes the following form:
+
+```json5
+{
+  "iss": "did:key:zStorageNode",
+  "aud": "did:key:zStorageNode",
+  "att": [
+    {
+      "with": "did:key:zStorageNode",
+      "can": "blob/replica/transfer",
+      "nb": {
+        /** The blob that will be transferred. */
+        "blob": {
+          "digest": { "/": { "bytes": "..." } },
+          "size": 1234
+        },
+        /** The location the blob will be transferred from. */
+        "location": { "/": "bafy..locationCommitment" },
```

Comment:
Is the intent for the node receiving the transfer invocation to "push" the data to the replicating node? Or is the intention for the replicating node to "pull" the data from the original node?

My read of both the `blob/replica/transfer` and `blob/replica/allocate` location field is that is specifies the location on the original node where the data resides. Is that right?